From 8977d120c37af3d145e4b8bd2393dd34a94b5143 Mon Sep 17 00:00:00 2001
From: Samuli Piippo <samuli.piippo@qt.io>
Date: Tue, 23 Oct 2018 09:54:57 +0300
Subject: [PATCH] Disable ltcg for windows and static libs

Disable ltcg when compiling on Windows host platform,
as GCC7 does not support lto with mingw.

Disable also for static builds since those libraries can be linked
on Windows host, where the lto built binaries won't work.

Task-number: QTBUG-71230
Change-Id: I5bea76ea03bf1943b0cfaf03d46a4840bd99a28b
Upstream-Status: Inappropriate [embedded specific]
---
 mkspecs/features/ltcg.prf | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mkspecs/features/ltcg.prf b/mkspecs/features/ltcg.prf
index 3cb01c334d..292e54235d 100644
--- a/mkspecs/features/ltcg.prf
+++ b/mkspecs/features/ltcg.prf
@@ -1,6 +1,6 @@
 static:no-static-ltcg {
     # Static library but no-static-ltcg enabled: skip LTCG
-} else: CONFIG(release, debug|release):!host_build {
+} else: CONFIG(release, debug|release):!host_build:!equals(QMAKE_HOST.os, Windows):!staticlib {
     # We need fat object files when creating static libraries on some platforms
     # so the linker will know to load a particular object from the library
     # in the first place. On others, we have special ar and nm to create the symbol
